tests/integration/WooCommerce/TransactionalEmailHooksTest.php

```diff
@@ -164,8 +164,8 @@ public function testUseTemplateForWCEmails() {
     );
     $transactionalEmails->useTemplateForWoocommerceEmails();
     expect($addedActions)->count(1);
-    expect($addedActions['woocommerce_init'])->callable();
-    $addedActions['woocommerce_init']();
+    expect($addedActions['woocommerce_email'])->callable();
+    $addedActions['woocommerce_email'](new \stdClass());
```

Comment:
Instead of passing the `stdClass` object, you can create a mock object. I think using a mock object is a better option because types will match and it will allow us using a type-hint for `WC_Emails`. You can use [Codeception methods for creating mocks](https://codeception.com/docs/reference/Mock) or just [PHPUnit mocks](https://phpunit.readthedocs.io/en/9.3/test-doubles.html#mock-objects)(link leads to the newest docs version but it seems that the API hasn't changed). In this case, I assume simple `$this->makeEmpty("\WC_Emails")` will do the job.
You will probably experience an issue with the non-existent class error. This is because we WooCommerce plugin is not loaded when we run integration tests. As a workaround, we currently create simple "stubs" for WooCommerce in [the bootstrap file](https://github.com/mailpoet/mailpoet/blob/master/tests/integration/_bootstrap.php#L245-L267). You can add WC_Emails there.

Reply:
Thanks, @costasovo. I had tried to create a mock object but stopped when I got the error saying that the class didn't exist. I wasn't sure where to create it. With your instructions, I manage to push a new commit with the changes that you suggested. When you get a chance, could you please review this PR again?